pkg/controller/dataimportcron-format-handlers.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2021 The CDI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+limitations under the License.
+See the License for the specific language governing permissions and
+*/
+
+package controller
+
+import (
+	"context"
+	"time"
+
+	snapshotv1 "github.com/kubernetes-csi/external-snapshotter/client/v6/apis/volumesnapshot/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	"kubevirt.io/containerized-data-importer/pkg/common"
+	cc "kubevirt.io/containerized-data-importer/pkg/controller/common"
+)
+
+var formatHandlers = map[cdiv1.DataImportCronSourceFormat]formatHandler{
+	cdiv1.DataImportCronSourceFormatSnapshot: snapshotFormatHandler{},
+	cdiv1.DataImportCronSourceFormatPvc:      pvcFormatHandler{},
+}
+
+type formatHandler interface {
+	populateDataSourceSpec(*cdiv1.DataSourceSpec, *cdiv1.DataVolumeSourcePVC)
+	handleSourceCreation(context.Context, *DataImportCronReconciler, *cdiv1.DataImportCron, *cdiv1.DataVolume, *storagev1.StorageClass) error
+	updateOnSuccess(context.Context, *DataImportCronReconciler, *cdiv1.DataImportCron, client.Object) error
```

Comment:
`updateOnSuccess` returns nil on all flows, sure you want it to return error?

Reply:
no reason to return error, thanks